server/new-committer.go

```diff
@@ -0,0 +1,180 @@
+		AuthorLink: fmt.Sprintf("https://github.com/%v", organisation),
+	}}
+
+	loadingPost := &model.Post{
+		ChannelId: args.ChannelId,
+		UserId:    p.botUserID,
+	}
+	model.ParseSlackAttachment(loadingPost, attachments)
+
+	loadingPost, appErr := p.API.CreatePost(loadingPost)
+	if appErr != nil {
+		return appErr
+	}
+
+	go p.updateNewCommittersPost(loadingPost, args.UserId, organisation, since)
+
+	return nil
+}
+
+func (p *Plugin) updateNewCommittersPost(post *model.Post, userID, org string, since time.Time) {
+	contributors, err := p.fetchContributors(org)
+	if err != nil {
+		p.API.LogError("failed to fetch data", "err", err.Error())
+
+		var message = "Failed to fetch data:" + err.Error()
+		if _, ok := err.(*github.RateLimitError); ok {
+			message = "Hit rate limit. Please try again later."
+		}
+		post.Props["attachments"].([]*model.SlackAttachment)[0].Text = message
+		p.updatePost(post, userID)
+		return
+	}
+
+	firstContributions := p.findFirstContributions(contributors)
+	p.filterContributionsByWeek(firstContributions, since)
+
+	var result []*firstContributionInfo
+	for committer, contribution := range firstContributions {
+		firstCommit := p.getFirstCommit(org, contribution.repo, committer, contribution.contributionWeek)
+		if firstCommit != nil && firstCommit.GetCommit().Committer.GetDate().After(since) {
+			result = append(result, &firstContributionInfo{firstCommit.GetAuthor().GetLogin(), firstCommit.GetCommit().Committer.GetDate(), firstCommit.GetHTMLURL(), org, contribution.repo})
+		}
+	}
+
+	sort.Slice(result, func(i, j int) bool {
+		return result[i].date.Before(result[j].date)
+	})
+
+	p.createPostContent(post, result, since)
+	p.updatePost(post, userID)
+}
+
+func (p *Plugin) createPostContent(post *model.Post, result []*firstContributionInfo, since time.Time) {
+	var resultText string
+	for _, e := range result {
+		resultText += fmt.Sprintf("- [%[1]s](https://github.com/%[1]v): [first commit](%s) at %s on [%[4]s](https://github.com/%s/%[4]v)\n", e.author, e.commit, e.date.Format(shortFormWithDay), e.repo, e.org)
+	}
+
+	attachment := post.Props["attachments"].([]*model.SlackAttachment)[0]
+	attachment.Title = "New Committers since " + since.Format(shortFormWithDay)
+	attachment.Text = ""
+	attachment.Fields = []*model.SlackAttachmentField{{
+		Title: "Number of new committers",
+		Value: strconv.Itoa(len(result)),
+	}, {
+		Title: "Committer",
+		Value: resultText,
+	}}
+}
+
+func (p *Plugin) updatePost(post *model.Post, userID string) {
+	if _, appErr := p.API.UpdatePost(post); appErr != nil {
+		p.SendEphemeralPost(post.ChannelId, userID, "Something went bad. Please try again. "+appErr.Where+" "+appErr.Id)
+		p.API.LogError("failed to update post", "err", appErr.Error())
+	}
+}
+
+func (p *Plugin) getFirstCommit(org, repo, committer string, week time.Time) *github.RepositoryCommit {
+	commits, err := p.fetchCommitsFromRepoByAuthorAndWeek(org, repo, committer, week)
+	if err != nil || len(commits) == 0 {
```

Comment:
I'm worried that returning `nil` if `p.fetchCommitsFromRepoByAuthorAndWeek` errored might bury an important error. Why not just return it?